pytorch_lightning/core/saving.py

```diff
@@ -247,6 +170,111 @@ def on_hpc_load(self, checkpoint: Dict[str, Any]) -> None:
         """
 
 
+def _load_from_checkpoint(
+    cls: Union["pl.LightningModule", "pl.LightningDataModule"],
+    checkpoint_path: Union[str, IO],
+    map_location: Optional[Union[Dict[str, str], str, torch.device, int, Callable]] = None,
+    hparams_file: Optional[str] = None,
+    strict: Optional[bool] = None,
+    **kwargs,
+):
+    with pl_legacy_patch():
+        if map_location is not None:
+            checkpoint = pl_load(checkpoint_path, map_location=map_location)
+        else:
+            checkpoint = pl_load(checkpoint_path, map_location=lambda storage, loc: storage)
+
+    if hparams_file is not None:
+        extension = hparams_file.split(".")[-1]
+        if extension.lower() == "csv":
+            hparams = load_hparams_from_tags_csv(hparams_file)
+        elif extension.lower() in ("yml", "yaml"):
+            hparams = load_hparams_from_yaml(hparams_file)
+        else:
+            raise ValueError(".csv, .yml or .yaml is required for `hparams_file`")
+
+        hparams["on_gpu"] = False
```

Comment:
I know this is copy-pasted, but why do we set this key in particular to False? Where does this come from?

Reply:
checked the git blame. Seems to be there since its inception. Should be removed since it's redundant now.